project_report/tex/analysis.tex

```diff
@@ -0,0 +1,168 @@
+\section{Analysis}
+
+\subsection{Use Case analysis}
+
+\subsubsection{Class Candidates}
+
+\subsubsection{Description of Classes}
+
+\subsubsection{UML Analysis Diagram}
+
+\subsection{Use Case Realisation}
+
+\subsubsection{Sequence Diagrams}
+
+            % \begin{enumerate}
+            %     \item A beer type should be selected
+            % \end{enumerate}
+
+\subsubsection{Operation Contracts}
```

Comment:
Maybe write something like:

An operation contract describes the responsibility of the operation. The contract focuses on what the operation can change, and not how it is changed. It is also used to describes the state of the system before and after the operation is called.